lib/src/functions/math.dart

```diff
@@ -289,11 +289,23 @@ final _random = math.Random();
 
 final _randomFunction = _function("random", r"$limit: null", (arguments) {
   if (arguments[0] == sassNull) return SassNumber(_random.nextDouble());
-  var limit = arguments[0].assertNumber("limit").assertInt("limit");
-  if (limit < 1) {
+  var limit = arguments[0].assertNumber("limit");
+
+  if (limit.hasUnits) {
+    warn(
+      "math.random() will no longer ignore \$limit units ($limit) in a "
+      "future release.\n"
+      "\n"
+      "If you meant to preserve units: "
+      "math.random($limit) * 1${limit.unitString}",
```

Comment:
Deprecation suggestions should generally be written so that they'll silence the warning, provide the desired behavior, *and* work after the breaking change. In this case, this will still produce a warning (since they're passing in the same limit) and it'll break once the unit is preserved (because the multiplication will add an extra unit).

A better suggestion might be: `math.random(\$limit / 1${limit.unitString}) * 1${limit.unitString})`.